### DISCUSSION TITLE
Use semantic versioning (MAJOR.MINOR.PATCH) for plugin API version

### DISCUSSION THREAD

**Discussion by pfroud** [OpenRCT2/OpenRCT2#21161] [Feature requests] [answered]
Currently, when the plugin API version is incremented, it might mean that a breaking change was made or a new feature was added. This can make it hard to predict what effect a new API version will have.

[Semantic Versioning](https://semver.org) would prevent this problem:

> Given a version number MAJOR.MINOR.PATCH, increment the:
>
>    - MAJOR version when you make incompatible API changes
>    - MINOR version when you add functionality in a backward compatible manner
>    - PATCH version when you make backward compatible bug fixes

For example, a breaking change is splitting `peep` into `guest` and `staff`. A non-breaking change is exposing the chain lift speed.

Changing the version scheme would itself create a new version, which would be confusing.

**Comment by ZehMatt** [comment]
I don't think it makes any difference, the plugin is supposed to specify what API version it uses and the game will actually keep the functionality for that specific version.

**Comment by Basssiiie** [comment, marked as answer]
Semantic versioning is useful for when you are the one deciding whether updating is safe. 

The plugin API version will just be incremented for users as they update the game, and they'll not know if it breaks plugins or not. The game, though, can handle backwards compatibility for plugins that target older versions. This should prevent breaking changes in the API from breaking older plugins.

It's very similar to how [Android's API levels](https://apilevels.com/) work.
